ziggurat-core-geoip/src/geoip.rs

```diff
@@ -0,0 +1,130 @@
+//use ipinfo::{IpInfo, IpInfoConfig};   // TODO(asmie): implement IpInfo provider
+use ip2location::{Record, DB};
+use ipgeolocate::{Locator, Service};
+
+/// List of supported geoip providers.
+pub enum Provider {
+    IpApiCom,
+    IpApiCo,
+    IpInfo,
+    Ip2Location,
+}
+
+/// Geoip provider service configuration.
+pub struct GeoIPService {
+    /// Geoip provider. Look at `Provider` enum for supported providers.
+    pub provider: Provider,
+    /// API key (paid or free) for online providers. For offline providers put here path to database file.
+    pub api_key: String,
+}
+
+/// IP information
+pub struct IPGeoInfo {
```

Comment:
let's use consistent naming:
- ziggurat-core-**geoip** 
- pub struct **GeoIP**Service {
```suggestion
pub struct GeoIPInfo {
```

or alternatively, we can omit `Geo` from struct names, because the user will call it with `geoip::IPInfo` - so no need to mention the `Geo` keyword twice